### PR TITLE
[FEATURE] Custom lnd data path

### PR DESCRIPTION
**Pull request by gamer90**
Add ability to set custom lnd data path.

### CODE REVIEW

**Comment by lverba** [comment]
rebuild this please

**Comment by lverba** [comment]
retest this please

File: frontend/components/restore/folder.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                            className={this.state.lndPathError ? "form-text__error" : ""}
+                            onChange={(e) => {
+                                if (e.target.files[0]) {
+                                    this.setState({ lndPath: e.target.files[0].path });
+                                } else {
+                                    this.setState({ lndPath: "" });
+                                }
+                            }}
+                        />
+                        <ErrorFieldTooltip text={this.state.lndPathError} />
```
[the remark this answers is not in the document]
Comment (gamer90) [reply frontend/components/restore/folder.js]: Added validation and field to state.

File: frontend/components/registration/reg-form.js [hunk 2]
```diff
@@ -194,6 +207,45 @@ class RegistrationForm extends PureComponent {
...
+                        </div>
+                    </div>
+                    <div className="col-xs-12">
+                        <File
+                            disabled={this.state.defaultPath}
+                            value={this.state.lndPath}
+                            placeholder="Select folder"
+                            className={this.state.lndPathError ? "form-text__error" : ""}
+                            onChange={(e) => {
+                                if (e.target.files[0]) {
```
[the remark this answers is not in the document]
Comment (gamer90) [reply frontend/components/registration/reg-form.js]: Fixed.

File: frontend/components/restore/user-form.js [hunk 3]
```diff
@@ -145,6 +158,47 @@ class UserForm extends Component {
...
+                            </Tooltip>
+                        </div>
+                    </div>
+                    <div className="col-xs-12">
+                        <File
+                            disabled={this.state.defaultPath}
+                            value={this.state.lndPath}
+                            placeholder="Select folder"
+                            className={this.state.lndPathError ? "form-text__error" : ""}
+                            onChange={(e) => {
```
[the remark this answers is not in the document]
Comment (gamer90) [reply frontend/components/restore/user-form.js]: Added validation.

File: frontend/components/ui/file.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+const File = ({
+    value, placeholder, disabled, className, ...props
+}) => (
+    <label className={`file__wrapper ${disabled ? "file__wrapper--disabled" : ""} ${className}`}>
```
[the remark this answers is not in the document]
Comment (gamer90) [reply frontend/components/ui/file.js]: Btn added.

File: frontend/components/restore/method.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    }
+}
+
+Method.propTypes = {
+    cancelRestore: PropTypes.func.isRequired,
+    useData: PropTypes.func.isRequired,
+    useSeed: PropTypes.func.isRequired,
+};
+
+const mapDispatchToProps = dispatch => ({
```
[the remark this answers is not in the document]
Comment (gamer90) [reply frontend/components/restore/method.js]: Fixed, but I think, dispatch actions/operations/etc better from `mapDispatchToProps` not from components, imho.

**Comment by lverba** [comment]
retest this please

**Comment by boodyvo** [comment]
Closed because is updated as [pr 28](https://github.com/LightningPeach/lightning-peach-wallet/pull/28).

**Pull request merged**